pkg/security/security.go

```diff
@@ -351,12 +352,42 @@ type Caller struct {
 	Identities []string
 }
 
+// AuthContext carries the contextual information and is passed across authenticators.
+type AuthContext struct {
+	// RequestContext is the context from request.
+	RequestContext context.Context
+	// Authenticators is the list of authenticators that were executed before in the auth chain.
+	Authenticators []string
+	// Caller indicates the caller that made authentication request.
+	Caller *Caller
+}
+
+// Authenticator determines the caller identity based on request context.
 type Authenticator interface {
-	Authenticate(ctx context.Context) (*Caller, error)
+	Authenticate(ctx *AuthContext) (*Caller, error)
 	AuthenticatorType() string
 	AuthenticateRequest(req *http.Request) (*Caller, error)
 }
 
+// DelegatedAuthenticator determines whether the main authenticator can trust the
+// client and proceed with authentication. This is used in cases where the client
+// authentication can not be directly determined from the request(no TLS certs, JWT etc)
+// but request has additional information to authenticate.
+// For example XfccAuthenticators should validate whether client is from trusted Cidr range
+// before it can trust the Xfcc Header.
+type DelegatedAuthenticator interface {
+	// CanTrustCaller indicates whether caller is a trusted caller.
+	CanTrustCaller(ctx *AuthContext) bool
+}
+
+func NewAuthContext(ctx context.Context) *AuthContext {
+	return &AuthContext{RequestContext: ctx}
+}
+
+func (ac *AuthContext) AddAuthenticator(authenticator string) {
```

Comment:
Why ? Each authenticator can add itself to the list in context ( and they can do that if they have anything to say - for example if there is no mTLS context, the mtls auth wouldn't add itself ). 

Also I think if we want to keep information about what authenticators where used - a map may be better, so each authenticator can also add its results. It's useful for debug and may be useful for further logic in Istiod.